### ISSUE TITLE
Crashes OBS on Version 2.2.2

### ISSUE THREAD

**Issue by MrTeeXD** [bozbez/win-capture-audio#123]
Hey there, I may have found a critical issue with the plugin that I want to bring to your attention. My guess is that, this crash occurs when a existing audio source is added and when OBS is relaunched, the plugin somehow cannot find the audio source?: 
```
Unhandled exception: c0000005
Date/Time: 2022-02-24, 22:05:21
Fault address: 200000018 ((null))
libobs version: 27.2.1 (64-bit)
Windows version: 10.0 build 19043 (release: 2009; revision: 1526; 64-bit)
CPU: Intel(R) Core(TM) i5-6400 CPU @ 2.70GHz


Thread 41AC: (Crashed)
Stack            EIP              Arg0             Arg1             Arg2             Arg3             Address
00000055B0AFF6D8 0000000200000018 0000000000000000 00000055B0AFF6B0 0000B6EF9E9DE1E1 0000000000000000 <unknown>!0x200000018
00000055B0AFF6E0 00007FFBF9758A56 00000000000007D8 000002402EF8C3C8 0000000000000000 00007FFBFEE8251D mmdevapi.dll!Ordinal14+0x1dcb6
00000055B0AFF710 00007FFBF97584DA 00000055B0AFF809 00007FFBF9758F66 00000055B3F5F830 00007FFB739644E6 mmdevapi.dll!Ordinal14+0x1d73a
00000055B0AFF740 00007FFBF9758684 000002402EF8C310 000000007FFE0301 0000000000000000 00000055B0AFF9E8 mmdevapi.dll!Ordinal14+0x1d8e4
00000055B0AFF770 00007FFBF9759088 000002402EF8C310 00007FFBF9720000 00000055B0AFF9E8 00000055B0AFF9E8 mmdevapi.dll!Ordinal14+0x1e2e8
00000055B0AFF7A0 00007FFBF97591E7 0000024029691180 0000000000000000 0000024029691180 0000024020E60B50 mmdevapi.dll!Ordinal14+0x1e447
00000055B0AFF820 00007FFC01332150 0000000000000000 0000000000000000 0000024020E60B50 0000024020ED1B40 ntdll.dll!0x7ffc01332150
00000055B0AFF870 00007FFC0132315A 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc0132315a
00000055B0AFFB70 00007FFC00E17034 0000000000000000 0000000000000000 0000000000000000 0000000000000000 kernel32.dll!0x7ffc00e17034
00000055B0AFFBA0 00007FFC01322651 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc01322651

Thread 41DC:
Stack            EIP              Arg0             Arg1             Arg2             Arg3             Address
00000055B06FC598 00007FFBFED8A104 0000024020E86DE0 0000024020ED1D70 0000000000000000 00007FF6F8391700 win32u.dll!0x7ffbfed8a104
00000055B06FC5A0 00007FFBFF45078E 0000024020ED1D70 0000005500000003 0000024020EADF60 0000024032969390 user32.dll!0x7ffbff45078e
00000055B06FC5E0 00007FFB7FC7ADEF 0000024020ED1D70 0000000000000000 0000024032969390 0000000000000000 qt5core.dll!0x7ffb7fc7adef
00000055B06FF700 00007FFB82212619 0000000000000000 00007FFB00000014 0000000000000000 0000000000000040 qwindows.dll!0x7ffb82212619
00000055B06FF730 00007FFB7FC2DE7C 0000024020E7F8A8 0000000000000000 00000055B06FFCD0 00007FFB7FD4FC90 qt5core.dll!0x7ffb7fc2de7c
00000055B06FF790 00007FFB7FC30994 00007FF6F8485100 00000055B06FFC10 00007FF6F8485100 00000055B06FFCD0 qt5core.dll!0x7ffb7fc30994
00000055B06FF7F0 00007FF6F81C86F4 0000000000000000 0000024020EA0E60 0000024000000000 0000024020E9E680 obs64.exe!run_program+0xa04
00000055B06FFBD0 00007FF6F81CA9C0 0000000000000000 0000000000000000 0000000000000000 FFFFFFFFFFFFFFFF obs64.exe!main+0x6f0
00000055B06FFDA0 00007FF6F8367707 0000000000000000 0000000000000000 0000000000000000 0000000000000000 obs64.exe!WinMain+0x157
00000055B06FFE30 00007FF6F8366FC6 0000000000000000 0000000000000000 0000000000000000 0000000000000000 obs64.exe!__scrt_common_main_seh+0x106
00000055B06FFE70 00007FFC00E17034 0000000000000000 0000000000000000 0000000000000000 0000000000000000 kernel32.dll!0x7ffc00e17034
00000055B06FFEA0 00007FFC01322651 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc01322651

Thread 41E0:
Stack            EIP              Arg0             Arg1             Arg2             Arg3             Address
00000055B07FF6F8 00007FFC013707C4 0000000000000000 0000000000000000 0000024020E6C6A0 0000024020E769A0 ntdll.dll!0x7ffc013707c4
00000055B07FF700 00007FFC01322DC7 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc01322dc7
00000055B07FFA00 00007FFC00E17034 0000000000000000 0000000000000000 0000000000000000 0000000000000000 kernel32.dll!0x7ffc00e17034
00000055B07FFA30 00007FFC01322651 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc01322651

Thread 41CC:
Stack            EIP              Arg0             Arg1             Arg2             Arg3             Address
00000055B08FF8A8 00007FFC013707C4 0000000000000000 0000000000000000 0000024020E6C6A0 0000024020E89920 ntdll.dll!0x7ffc013707c4
00000055B08FF8B0 00007FFC01322DC7 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc01322dc7
00000055B08FFBB0 00007FFC00E17034 0000000000000000 0000000000000000 0000000000000000 0000000000000000 kernel32.dll!0x7ffc00e17034
00000055B08FFBE0 00007FFC01322651 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc01322651

Thread 41C4:
Stack            EIP              Arg0             Arg1             Arg2             Arg3             Address
00000055B09FF658 00007FFC013707C4 0000000000000000 0000000000000000 0000024020E6C6A0 0000024020E8EC30 ntdll.dll!0x7ffc013707c4
00000055B09FF660 00007FFC01322DC7 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc01322dc7
00000055B09FF960 00007FFC00E17034 0000000000000000 0000000000000000 0000000000000000 0000000000000000 kernel32.dll!0x7ffc00e17034
00000055B09FF990 00007FFC01322651 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc01322651

Thread 4200:
Stack            EIP              Arg0             Arg1             Arg2             Arg3             Address
00000055B0BFF7B8 00007FFC013707C4 0000024020E60C68 0000024020E60C68 0000024020E60B50 0000024020E60B68 ntdll.dll!0x7ffc013707c4
00000055B0BFF7C0 00007FFC01322DC7 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc01322dc7
00000055B0BFFAC0 00007FFC00E17034 0000000000000000 0000000000000000 0000000000000000 0000000000000000 kernel32.dll!0x7ffc00e17034
00000055B0BFFAF0 00007FFC01322651 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc01322651

Thread 41B8:
Stack            EIP              Arg0             Arg1             Arg2             Arg3             Address
00000055B0CFFBF8 00007FFC0136CDF4 0000000000000000 00000240328C2390 0000000000000000 0000000000000000 ntdll.dll!0x7ffc0136cdf4
00000055B0CFFC00 00007FFBFEE41A5E 0000000000007530 0000000000007530 0000024000000000 000000000000046C kernelbase.dll!0x7ffbfee41a5e
00000055B0CFFCA0 00007FFB7FA8B11B 0000024020EDFCD0 0000024020EDFCE0 0000024020EAF2A8 0000000000000001 qt5core.dll!0x7ffb7fa8b11b
00000055B0CFFCF0 00007FFB7FA8B186 0000024020EAF2A9 0000024020EAF2A8 0000024020EDFCD0 0000000000000000 qt5core.dll!0x7ffb7fa8b186
00000055B0CFFD20 00007FFB7FA893B3 0000024020EDFCD0 0000024020EAF2A9 0000024020EDFCD0 0000024020E7F3A8 qt5core.dll!0x7ffb7fa893b3
00000055B0CFFDB0 00007FFB7FA85FF1 0000000000000000 0000000000000000 0000000000000000 0000000000000000 qt5core.dll!0x7ffb7fa85ff1
00000055B0CFFDE0 00007FFC00E17034 0000000000000000 0000000000000000 0000000000000000 0000000000000000 kernel32.dll!0x7ffc00e17034
00000055B0CFFE10 00007FFC01322651 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc01322651

Thread 4224:
Stack            EIP              Arg0             Arg1             Arg2             Arg3             Address
00000055B0DFFB68 00007FFC0136CDF4 0000000000000000 00000240328C2390 0000000000000000 0000000000000000 ntdll.dll!0x7ffc0136cdf4
00000055B0DFFB70 00007FFBFEE41A5E 0000000000007530 0000000000007530 0000024000000000 0000000000000470 kernelbase.dll!0x7ffbfee41a5e
00000055B0DFFC10 00007FFB7FA8B11B 0000024020EDFB80 0000024020EDFB90 0000024020EAF2A8 0000000000000001 qt5core.dll!0x7ffb7fa8b11b
00000055B0DFFC60 00007FFB7FA8B186 0000024020EAF2A9 0000024020EAF2A8 0000024020EDFB80 0000000000000000 qt5core.dll!0x7ffb7fa8b186
00000055B0DFFC90 00007FFB7FA893B3 0000024020EDFB80 0000024020EAF2A9 0000024020EDFB80 0000024020E7FDA8 qt5core.dll!0x7ffb7fa893b3
00000055B0DFFD20 00007FFB7FA85FF1 0000000000000000 0000000000000000 0000000000000000 0000000000000000 qt5core.dll!0x7ffb7fa85ff1
00000055B0DFFD50 00007FFC00E17034 0000000000000000 0000000000000000 0000000000000000 0000000000000000 kernel32.dll!0x7ffc00e17034
00000055B0DFFD80 00007FFC01322651 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc01322651

Thread 41A0:
Stack            EIP              Arg0             Arg1             Arg2             Arg3             Address
00000055B0EFFB68 00007FFC0136CDF4 0000000000000000 00000240328C2390 0000000000000000 0000000000000000 ntdll.dll!0x7ffc0136cdf4
00000055B0EFFB70 00007FFBFEE41A5E 0000000000007530 0000000000007530 0000024000000000 0000000000000474 kernelbase.dll!0x7ffbfee41a5e
00000055B0EFFC10 00007FFB7FA8B11B 0000024020EDFFA0 0000024020EDFFB0 0000024020EAF2A8 0000000000000001 qt5core.dll!0x7ffb7fa8b11b
00000055B0EFFC60 00007FFB7FA8B186 0000024020EAF2A9 0000024020EAF2A8 0000024020EDFFA0 0000000000000000 qt5core.dll!0x7ffb7fa8b186
00000055B0EFFC90 00007FFB7FA893B3 0000024020EDFFA0 0000024020EAF2A9 0000024020EDFFA0 0000024020E7FE48 qt5core.dll!0x7ffb7fa893b3
00000055B0EFFD20 00007FFB7FA85FF1 0000000000000000 0000000000000000 0000000000000000 0000000000000000 qt5core.dll!0x7ffb7fa85ff1
00000055B0EFFD50 00007FFC00E17034 0000000000000000 0000000000000000 0000000000000000 0000000000000000 kernel32.dll!0x7ffc00e17034
00000055B0EFFD80 00007FFC01322651 0000000000000000 0000000000000000 0000000000000000 0000000000000000 ntdll.dll!0x7ffc01322651

Thread 419C:
Stack            EIP              Arg0             Arg1             Arg2             Arg3             Address
00000055B0FFFB58 00007FFC0136CDF4 0000000000000000 00000240328C2390 0000000000000000 0000000000000000 ntdll.dll!0x7ffc0136cdf4
```

**Comment by georgiddam** [comment]
I have the exact same issue with (i'm with) V2.1.0 and i've had it for a while, just realised its most probably from this plugin.

**Comment by mazunki** [comment]
@MrTeeXD Did you figure out what causes the crash
?

**Comment by MrTeeXD** [comment]
Sadly no, Its unclear how this plugin crashes OBS @mazunki

**Comment by MrTeeXD** [comment]
Also, I did not close this issue.

**Comment by mazunki** [comment]
Thanks, it would probably be helpful if you also shared the logs, not only the coredump. My friend is having the same issue, and also uses this plugin.

**Comment by MrTeeXD** [comment]
With a Plugin like this, its impossible to reproduce the crash. I think only the Dev can answer this.

**Comment by bozbez** [comment]
@mazunki does your friend have an issue with their logs in? There are more duplicates of this than I care to search...

**Comment by MrTeeXD** [comment]
I can close my other Duplicate and consolidate the other one into this one

**Comment by MrTeeXD** [comment]
And I agree with Bozbez, if people continue to submit duplicates, it WILL cause alot of confusion. There is the Discord Server for this plugin but people don't bother going there.

**Comment by MrTeeXD** [comment]
Apparently, this `mmdevapi.dll!Ordinal14+0x1dcb6` keeps popping up everytime it crashes.

Here is what I did that I think could have made the plugin crash:
1- Open OBS and the Application
2- Added the Plugin to the Sources List then add the Application in the Plugin
3- Closed OBS but keep the Application open
4- Re-open OBS while the plugin is added in the Sources list
5- It should crash???

**Comment by vkryvor** [comment]
@MrTeeXD did you figure out what is causing this? I am experiencing the same issue

**Comment by ghost** [comment]
To reduce the amount of stress towards the Dev, I closed this as I believe this to be an OS-Leveled Issue.

**Comment by mazunki** [comment]
Even if the cause for the bug here is upstream, the bug still persists. I'd keep the bug report open, unless the dev requests a cleanup. Perhaps there's something we can do from our end to bypass the upstream bug, if that's the case.

I saw it had been closed, and was about to tell my friend they had fixed it. PS: I don't have a Windows computer to provide any logs or coredumps, so sorry for not being able to provide them.
